processors/filterprocessor/expression/expression.go

```diff
@@ -59,6 +83,69 @@ func MakeExpression(expressionType FilterType, expressionSearchStr string) (*Exp
 			var exp Expression = regexExpression{Regex: r}
 			return &exp, nil
 		}
+
+	case LessThanFilter:
+		fallthrough
+	case LessThanEqualFilter:
+		fallthrough
+	case GreaterThanFilter:
+		fallthrough
+	case GreaterThanEqualFilter:
+		fallthrough
+	case EqualToFilter:
+		fallthrough
+	case NotEqualToFilter:
+		{
+			var exp Expression
+
+			switch targetKind {
+			case reflect.TypeOf(basics.MicroAlgos{}).Kind():
```

Comment:
You could put this check in the generator. If the leaf type is MicroAlgos add the extra Raw reference:
```diff
	case "txn.fee":
-		return &input.SignedTxnWithAD.SignedTxn.Txn.Header.Fee, nil
+		return &input.SignedTxnWithAD.SignedTxn.Txn.Header.Fee.Raw, nil
```